### PR TITLE
Added GKE create/delete functionality, updated README and improved id…

### PR DESCRIPTION
**Pull request by oliver-reload**
…empotency of docker container flow

### CODE REVIEW

File: .github/workflows/containers.yml [hunk 1]
```diff
@@ -9,6 +9,8 @@ jobs:
 
     steps:
     - uses: actions/checkout@v2
-    - name: Execute gc.sh
-      run: ./gc.sh
+    - name: Execute scripts
```
Comment (achton) [inline .github/workflows/containers.yml]: Title a little generic for my taste :)

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -14,5 +14,5 @@ jobs:
 
     - uses: actions/checkout@v2
 
-    - name: Check our script for errors
-      run: shellcheck ./containers.sh ./gc.sh
+    - name: Check our scripts for errors
+      run: shellcheck ./docker_functions.sh ./create_containers.sh ./gc.sh
```
Comment (achton) [inline .github/workflows/tests.yml]: Is it possible to do this perhaps?

```suggestion
      run: shellcheck ./*.sh
```

File: create_containers.sh [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+source ./docker_functions.sh
+
+# Create networks for containers
+log "Checking for existing networks..."
+for network in "${networks[@]}"
+do
+  if [ "$(docker network ls | grep -w "${network}")" ]; then
+    log "Network ${network} exists, skipping..."
+  else
+    OUTPUT=$(networks)
```
Comment (achton) [inline create_containers.sh]: The naming becomes confusing here, should probably be `create_networks()` or something like that to signify what happens. Same for some of the other functions I think.

Also $OUTPUT is never used?

File: create_containers.sh [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    OUTPUT=$(volumes)
+    log "Volume ${volume} created"
+  fi
+done
+
+# Spawn todo list node.js app
+if [ "$(docker ps -a | grep -w "${app}")" ]; then
+  # cleanup
+  log "Existing ${app} container found - nuking & re-spawning it"
+  docker rm -f "${app}" > /dev/null
```
Comment (achton) [inline create_containers.sh]: In most cases, we want the output available unless there is a specific reason or bug requiring that it is hidden.

File: create_gke_cluster.sh [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+source ./gke_functions.sh
+
+log "Configuring gcloud cli for cluster: ${GKE_CLUSTER_NAME}..."
+OUTPUT=$(config_gcloud)
+
+log "Creating GKE cluster: ${GKE_CLUSTER_NAME}"
+OUTPUT=$(gke_create)
+log "${GKE_CLUSTER_NAME} scheduled for creation (~20 min.)"
+log "Remember to configure kubectl locally when cluster is available using: gcloud container clusters get-credentials ${GKE_CLUSTER_NAME}"
```
Comment (achton) [inline create_gke_cluster.sh:13]: Nice :+1:

**Review by achton — CHANGES_REQUESTED** [review]
A few minor things, but looks good so far :+1:

File: create_containers.sh (hunk 3 above)
Comment (oliver-reload) [reply create_containers.sh]: I have fixed my naming convention, but I'm not sure I follow your comment regarding OUTPUT never being used:
```
oliver@skynet:~/reload/onboarding-oliver$ bash create_containers.sh 
09:52:25 Checking for existing networks...
09:52:25 Network todo-app created
oliver@skynet:~/reload/onboarding-oliver$ bash create_containers.sh 
09:53:42 Checking for existing networks...
09:53:42 Network todo-app exists, skipping...
```

**Pull request merged**